### DISCUSSION TITLE
Is there a way to pull only open issues/PRs?

### DISCUSSION THREAD

**Discussion by philn** [magit/forge#693] [Support] [answered]
For a big gitlab project with thousands of closed MRs, the initial pull is endless :(

What is the point of pulling closed topics?

**Comment by tarsius** [comment, marked as answer]
> What is the point of pulling closed topics?

Sometimes you have to go back and read why something was done the way it was done. ;D

> Is there a way to pull only open issues/PRs?

No, but after updating to `v0.4.0` (or better `v0.4.1`), you are given this choice

```
Adding XYX to database,
 a pulling all topics
 s pulling only topics since <date>
 i to allow pulling of individual topics
```

and should press `s`, to pick a date and only have topics pulled that have been modified since that date (but that will also include closed topics).

**Reply by philn** [reply]
Thanks :)

I think that if I wanted to visit a closed topic I suppose I would pull it first (opt-in), somehow :)

I don't know how this works, so please forgive my naive question :) Would it be possible to pull only open topics by default but somehow keep the list (not the full contents) of closed topics locally?

**Reply by tarsius** [reply]
> Would it be possible to pull only open topics by default but somehow keep the list (not the full contents) of closed topics locally?

Theoretically everything is possible but

- Fetching from Gitlab is currently callback hell. Until that has been refactored, I won't consider adding alternative code-paths.
- A lot of code currently assumes that a topic either exists "fully" locally or not at all, and would need adjustments.
- We could not fetch a topics posts, but would have to fetch everything else. And I think (but haven't checked) that would mean we wouldn't avoid all that much time by not fetching the posts.

**Reply by philn** [reply]
Alright, I understand, thanks for taking the time to explain this. I think that for the time being I'll keep suffering with the gitlab web UI then. :(

**Reply by tarsius** [reply]
Someday Gitlab support will improve ... see future release announcement. Will likely take a while, because it is as much work as Github support, and there are waaaaay more Github users, so my motivation is somewhat limited. (Especially because this is just a stepping stone anway; I would very much like to move to a Git-native solution, without a forge complicating matters.)
